sys/log/full/src/log_fcb.c

```diff
@@ -932,12 +933,21 @@ log_fcb_copy_entry(struct log *log, struct fcb_entry *entry,
         goto err;
     }
 
-    /* Changing the fcb to be logged to be dst fcb */
-    fcb_tmp = &((struct fcb_log *)log->l_arg)->fl_fcb;
+    /* Cache fcb_log pointer */
+    fcb_log_ptr = (struct fcb_log *)log->l_arg;
 
-    log->l_arg = dst_fcb;
+    /* Cache the fcb log, so that we preserve original fcb pointer and
+     * bookmark settings
```

Comment:
Comment suggest that bookmark settings need to be preserved.
It bookmarks can be affected by log_fcb_append() only part of bookmark data is preserved.
log_fcb_bset as part of fcb_log would be preserved while elements of lfs_bmarks array could be modified.
Finally fcl_log would have data intact but not in sync with bookmark array that could have pointer(s) to scratch area.

Reply:
The bookmarks do get rotated later on or re-inited incase of sector bookmarks. so, it is fine I think. I just removed the comment.